frontend/src/components/dialogs/WebterminalServiceAccountDialog.vue

```diff
@@ -0,0 +1,174 @@
+          <span v-else>To access the garden cluster a dedicated service account is required.</span>
+        </div>
+        <div>
+           <span v-if="needsUpdate">Do you want grant the <tt>admin</tt> role to the service account?</span>
+           <span v-else>Do you want to create the <tt>{{serviceAccountName}}</tt> service account and add it as member with <tt>admin</tt> role to this project?</span>
+          <v-list>
+            <v-list-item>
+              <v-list-item-content>
+                <v-list-item-title class="wrap-text">
+                  <account-avatar :account-name="serviceAccountUsername"></account-avatar>
+                </v-list-item-title>
+              </v-list-item-content>
+              <v-list-item-action>
+                <service-account-roles :role-display-names="desiredRoleDisplayNames"></service-account-roles>
+              </v-list-item-action>
+            </v-list-item>
+          </v-list>
+        </div>
+      </div>
+      <g-alert
+        color="error"
+        class="ma-0"
+        :message.sync="errorMessage"
+        :detailedMessage.sync="detailedErrorMessage"
+      ></g-alert>
+    </template>
+  </g-dialog>
+</template>
+
+<script>
+import GDialog from '@/components/dialogs/GDialog'
+import AccountAvatar from '@/components/AccountAvatar'
+import GAlert from '@/components/GAlert'
+import ServiceAccountRoles from '@/components/ServiceAccountRoles'
+import { errorDetailsFromError, isConflict } from '@/utils/error'
+import { mapActions } from 'vuex'
+import get from 'lodash/get'
+import { sortedRoleDisplayNames } from '@/utils'
+
+export default {
+  name: 'WebterminalServiceAccountDialog',
+  components: {
+    GDialog,
+    GAlert,
+    AccountAvatar,
+    ServiceAccountRoles
+  },
+  props: {
+    namespace: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      errorMessage: undefined,
+      detailedErrorMessage: undefined,
+      member: undefined
+    }
+  },
+  computed: {
+    needsUpdate () {
+      return !!this.member
+    },
+    confirmButtonText () {
+      if (this.needsUpdate) {
+        return 'Update'
+      }
+      return 'Add'
+    },
+    serviceAccountName () {
+      return 'dashboard-webterminal'
+    },
+    serviceAccountUsername () {
+      return `system:serviceaccount:${this.namespace}:${this.serviceAccountName}`
+    },
+    desiredRoleDisplayNames () {
+      return sortedRoleDisplayNames(this.desiredRoles)
+    },
+    desiredRoles () {
+      const roles = [...get(this.member, 'roles', [])]
+      roles.push('admin')
+      return roles
+    }
+  },
+  methods: {
+    ...mapActions([
+      'addMember',
+      'updateMember'
+    ]),
+    async promptForConfirmation (member) {
+      this.member = member
+
+      return await this.$refs.gDialog.confirmWithDialog(async () => {
+        return this.addServiceAccount()
+      })
+    },
+    async addServiceAccount () {
+      if (!this.namespace) {
+        console.error('no namespace set')
```

Comment:
if we really need it
```suggestion
        // eslint-disable-next-line no-console
        console.error('no namespace set')
```

Reply:
this is a safetey measure to prevent adding service accounts as project member in case namespace is undefined. Otherwise a service account lie `system:serviceaccount:undefined:foo` would be added. Someone owing the `undefined` namespace could then gain access to the project.